spec/lib/dfe/bigquery/application_metrics_spec.rb

```diff
@@ -232,4 +232,184 @@
       expect(application_metrics.last.subject_filter).to eq('Secondary')
     end
   end
+
+  describe '.route_into_teaching' do
+    subject(:application_metrics) do
+      described_class.route_into_teaching(cycle_week: 7)
+    end
+
+    let(:results) do
+      [
+        {
+          nonsubject_filter: 'Postgraduate teaching apprenticeship',
+          cycle_week: 7,
+        },
+        {
+          nonsubject_filter: 'School Direct (salaried)',
+          cycle_week: 7,
+        },
+      ]
+    end
+
+    before do
+      allow(client).to receive(:query)
+        .with(
+          <<~SQL,
+            SELECT *
+            FROM dataform.application_metrics
+            WHERE recruitment_cycle_year = 2024
+            AND cycle_week = 7
+            AND subject_filter_category = "Total excluding Further Education"
+            AND nonsubject_filter_category = "Route into teaching"
+          SQL
+        )
+        .and_return(results)
+    end
+
+    it 'instantiate an application metrics' do
+      expect(application_metrics).to be_instance_of(Array)
+      expect(application_metrics.size).to be 2
```

Comment:
This is a pattern used throughout the spec file, but I don't think it's the expectaion you'd want here. I'm not sure what the most straightforward method for comparing these would be, but what do you think of something like
```
it 'returns the correct results' do
  expect(application_metrics.as_json).to eq(results.as_json)
end
```

Reply:
good suggestion. Done